src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Concurrent;
+using Microsoft.Framework.DependencyInjection;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class DefaultViewComponentFactory : IViewComponentFactory
+    {
+        private readonly IServiceProvider _provider;
+        private static readonly Func<Type, ObjectFactory> CreateFactory =
+            (t) => ActivatorUtilities.CreateFactory(t, Type.EmptyTypes);
+        private static readonly ConcurrentDictionary<Type, ObjectFactory> _viewComponentCache =
+               new ConcurrentDictionary<Type, ObjectFactory>();
+
+        public DefaultViewComponentFactory([NotNull] IServiceProvider provider)
```

Comment:
this isn't good. This is another case where caching can't work. We shouldn't see serviceprovider in the constructor of anything that performs caching.